presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTableHandle.java

```diff
@@ -79,6 +86,12 @@ public TableType getTableType()
         return predicate;
     }
 
+    @JsonProperty
+    public TupleDomain<IcebergColumnHandle> getEnforcedPredicate()
```

Comment:
What are the semantics of `IcebergTableHandle.getEnforcedPredicate`? Is it guaranteed to be a tight bound?

Reply:
In particular how do `predicate` and `enforcedPredicate` differ from each other. From my reading of the above code, this is just about whether the predicate is over the partition columns vs non-partition columns. 

Maybe it'd make more sense to model it as two complementary sets of predicates (partitionColumnPredicate vs nonPartitionColumnPredicate). It would also make it easier to interpret and understand all the places where they are referenced.

Reply:
@martint `enforcedPredicate` - predicates that can be enforced by the iceberg connector because they are over **identity** partition columns that are present in **all** partition specs. predicates on other columns are put in `predicate`. They're complementary currently. 

Given the nuances mentioned above, didn't name them as "partitionColumnPredicate" as it's inaccurate.  should we rename `predicate` as `unEnforcedPredicate` instead? we can say `unEnforcedPredicate` are not guaranteed to be enforced, but still being used for split-generation and reader-level optimizations.

Reply:
Yes, that would be better.